code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -529,6 +529,11 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 					user.visible_message("<span class='notice'>After a few attempts, [user] manages to light the [src].</span>")
 				else
 					user << "<span class='warning'>You burn yourself while lighting the lighter.</span>"
+					var/not_fire = !user.on_fire
+					user.IgniteMob()
+					if(not_fire && user.on_fire)

```

Comment:
This. Seems like a really weird way of handling this shit.

Either make `IgniteMob` return a meaningful value or make `IgniteMob` generate the messages. The proposed method is bad logic.